src/librustc_resolve/lib.rs

```diff
@@ -797,36 +787,103 @@ bitflags! {
     }
 }
 
-// Records a possibly-private type definition.
+// Records a possibly-private definition.
 #[derive(Clone,Debug)]
-struct TypeNsDef {
+struct NsDef {
     modifiers: DefModifiers, // see note in ImportResolution about how to use this
-    module_def: Option<Rc<Module>>,
-    type_def: Option<Def>,
-    type_span: Option<Span>
+    def: Option<Def>,
+    module: Option<Rc<Module>>,
```

Comment:
Some comments here would be good. e.g. what does it mean if both `def` and `module` are `Some`? The function `def` below seems to suggest that `def` "takes precedence" -- does this mean that `module` should always be `None` when `def` is `Some`?